apps/dashboard/src/app/(dashboard)/(chain)/[chain_id]/[contractAddress]/modules/components/OpenEditionMetadata.tsx

```diff
@@ -0,0 +1,220 @@
+    isOwnerAccount: boolean;
+    setSharedMetadata: (values: SetSharedMetadataFormValues) => Promise<void>;
+  },
+) {
+  return (
+    <ModuleCardUI {...props}>
+      <div className="h-1" />
+      <Accordion type="single" collapsible className="-mx-1">
+        <AccordionItem value="metadata" className="border-none">
+          <AccordionTrigger className="border-border border-t px-1">
+            Set Shared Metadata
+          </AccordionTrigger>
+          <AccordionContent className="px-1">
+            {props.isOwnerAccount && (
+              <SetSharedMetadataSection
+                setSharedMetadata={props.setSharedMetadata}
+              />
+            )}
+            {!props.isOwnerAccount && (
+              <Alert variant="info">
+                <CircleAlertIcon className="size-5" />
+                <AlertTitle>
+                  You don't have permission to set shared metadata on this
+                  contract
+                </AlertTitle>
+              </Alert>
+            )}
+          </AccordionContent>
+        </AccordionItem>
+      </Accordion>
+    </ModuleCardUI>
+  );
+}
+
+function SetSharedMetadataSection(props: {
+  setSharedMetadata: (values: SetSharedMetadataFormValues) => Promise<void>;
+}) {
+  const form = useForm<SetSharedMetadataFormValues>({
+    resolver: zodResolver(formSchema),
+    values: {
+      name: "",
+      description: "",
+      imageUri: "",
+      animationUri: "",
+    },
+    reValidateMode: "onChange",
+  });
+
+  const setSharedMetadataMutation = useMutation({
+    mutationFn: props.setSharedMetadata,
+  });
+
+  const onSubmit = async () => {
+    const promise = setSharedMetadataMutation.mutateAsync(form.getValues());
+    toast.promise(promise, {
+      success: "Successfully set shared metadata",
+      error: (error) => `Failed to set shared metadata: ${error}`,
+    });
+  };
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)}>
+        <div className="flex flex-col gap-6">
+          <div className="flex gap-4">
+            <FormField
+              control={form.control}
+              name="name"
+              render={({ field }) => (
+                <FormItem className="flex-1">
+                  <FormLabel>Name</FormLabel>
+                  <FormControl>
+                    <Input {...field} />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+            <FormField
+              control={form.control}
+              name="description"
+              render={({ field }) => (
+                <FormItem className="flex-1">
+                  <FormLabel>Description</FormLabel>
+                  <FormControl>
+                    <Input {...field} />
+                  </FormControl>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+          </div>
+
+          <div className="flex gap-4">
```

Comment:
Make all the inputs full width on mobile here, change the submit button text to "Update"  

<img width="467" alt="image" src="https://github.com/user-attachments/assets/5a1f2ec0-ae75-4cf1-b4e7-907b1120c221">